src/scales/scale.timeseries.js

```diff
@@ -0,0 +1,190 @@
+					helpers.each(dataset.data, function(value, dataIndex) {
+						var dataMoment = timeHelpers.parseTime(me, me.getRightValue(value));
+
+						if (dataMoment.isValid()) {
+							if (timeOpts.round) {
+								dataMoment.startOf(timeOpts.round);
+							}
+
+							timestamps[dataIndex] = dataMoment.valueOf();
+						}
+					});
+				} else {
+					// We have no x coordinates, so use the ones from the labels
+					timestamps = parsedData.labels.slice();
+				}
+
+				parsedData.datasets[datasetIndex] = timestamps;
+			});
+
+			var allTimestamps = parsedData.labels;
+			helpers.each(parsedData.datasets, function(value) {
+				allTimestamps = allTimestamps.concat(value);
+			});
+
+			allTimestamps = arrayUnique(allTimestamps).sort(function(a, b) {
+				return a - b;
+			});
+
+			parsedData.allTimestamps = allTimestamps;
+
+			me._parsedData = parsedData;
+		},
+
+		buildTicks: function() {
+			var me = this;
+			var timeOpts = me.options.time;
+
+			var allTimestamps = me._parsedData.allTimestamps;
+			var dataMin = allTimestamps[0];
+			var dataMax = allTimestamps[allTimestamps.length - 1];
+
+			var maxTicks = me.getLabelCapacity(dataMin);
+			var unit = timeOpts.unit || timeHelpers.determineUnit(timeOpts.minUnit, dataMin, dataMax, maxTicks);
+			me.displayFormat = timeOpts.displayFormats[unit];
+
+			me._tickTimestamps = allTimestamps;
+			me.ticks = allTimestamps;
+		},
+
+		// Used to get data value locations.  Value can either be an index or a numerical value
+		getPixelForValue: function(value, index, datasetIndex, includeOffset) {
+			var me = this;
+
+			var offsetAmt = Math.max((me.ticks.length - ((me.options.gridLines.offsetGridLines) ? 0 : 1)), 1);
+
+			if (typeof datasetIndex === 'number') {
+				var timestamp = me._parsedData.datasets[datasetIndex][index];
+				var indexByTimestamp = me._tickTimestamps.indexOf(timestamp);
+				index = indexByTimestamp !== -1 ? indexByTimestamp : index;
+			}
+
+			var valueWidth = me.width / offsetAmt;
+			var widthOffset = valueWidth * index;
+
+			if (me.options.gridLines.offsetGridLines && includeOffset || me.ticks.length === 1 && includeOffset) {
+				widthOffset += (valueWidth / 2);
+			}
+
+			return me.left + Math.round(widthOffset);
+		},
+		getPixelForTick: function(index, includeOffset) {
+			if (this.ticks.length === 1) {
+				includeOffset = true;
+			}
+			return this.getPixelForValue(this.ticks[index], index, null, includeOffset);
+		},
+		getValueForPixel: function(pixel) {
+			var me = this;
+			var value;
+			var offsetAmt = Math.max((me.ticks.length - ((me.options.gridLines.offsetGridLines) ? 0 : 1)), 1);
+			var valueDimension = me.width / offsetAmt;
+
+			pixel -= me.left;
+
+			if (me.options.gridLines.offsetGridLines) {
+				pixel -= (valueDimension / 2);
+			}
+
+			if (pixel <= 0) {
+				value = 0;
+			} else {
+				value = Math.round(pixel / valueDimension);
+			}
+
+			return value;
+		},
+		getBasePixel: function() {
+			return this.bottom;
```

Comment:
what will this do for horizontal bar charts using the timeseries axis as the y axis?